### ISSUE TITLE
$GLOBALS['TL_BODY'] does not work in Backend

### ISSUE THREAD

**Issue by Fuzzyma** [contao/core#8179]
Stuff you add to the array is not getting displayed at the end of the body. Actually `$GLOBALS['TL_BODY']` is not even an array.

**Comment by Aybee** [comment]
I can not confirm this. `$GLOBALS['TL_BODY']` expects to be an array. So you have to fill it like this
```php
$GLOBALS['TL_BODY'][] = '<script src="files/js/myLateLoad.js"></script>';
```

**Comment by Fuzzyma** [comment]
> Actually $GLOBALS['TL_BODY'] is not even an array.

I var_dumped it - its null.

> ```php
$GLOBALS['TL_BODY'][] = '<script src="files/js/myLateLoad.js"></script>';
```

Thats how I did it. Script is not loaded neither is the code displayed in the html output.

**Comment by Aybee** [comment]
Where did you write your code? In dcaconfig.php or elsewhere?

**Comment by Fuzzyma** [comment]
In my widget which requires custom Javascript.
I dont want to use TL_JAVASCRIPT because I want to load the script at the bottom of the page

**Comment by BugBuster1701** [comment]
then try this, like mootools:
```php
$GLOBALS['TL_MOOTOOLS '][] = "<script type='text/javascript'>....  ";
```

**Comment by Fuzzyma** [comment]
Sorry to offend you but I wrote this issue to fix the bug (when it is a bug), that TL_BODY is not working.
I am aware that I can use TL_MOOTOOLS but it remains a hack. I would like to use the correct type

**Comment by Aybee** [comment]
You don't offend me. Just wondering what you are doing wrong. It works, when I put it in 
- system/dcaconfig.php
- a config.php of a module (extension)
- a template
- a file included with {{file::myfile.html5}}

That's why I asked where you put your code in. I don't know your "widget".

**Comment by Fuzzyma** [comment]
Its a normal backend widget. In its constructor (or in the generate method - i tried both) I wrote the code.
Nothing happened.

```php
class FormAdresspickerField extends FormTextField
{
    /**
     * Template
     *
     * @var string
     */
    //protected $strTemplate = 'form_adresspicker';
    protected $strTemplate = 'be_widget';
    private $options = array();

    public function __construct($arrAttributes = null)
    {
        if(TL_MODE == 'FE'){
            $strTemplate = 'form_widget';
        }
    
        parent::__construct($arrAttributes);
        
        $this->evals = $arrAttributes;
        
        $GLOBALS['TL_BODY'][] = '<script... ...>';

    }

// ...
}

```

Maybe the code helps

**Comment by Aybee** [comment]
Ah, ok. I just tested in FE. Don't know, if TL_BODY works in BE.

**Comment by fritzmg** [comment]
@Fuzzyma have you actually checked if `TL_MOOTOOLS` works? As far as I can see in the code, neither should work in the backend, because it's only handled in [`\Controller::replaceDynamicScriptTags`](https://github.com/contao/core/blob/3.5.6/system/modules/core/library/Contao/Controller.php#L697) which in turn is only called in [`\FrontendTemplate`](https://github.com/contao/core/blob/3.5.6/system/modules/core/classes/FrontendTemplate.php).

**Comment by Fuzzyma** [comment]
No I didnt check it. I was quite sure, that this will work (since mootools is deeply anchored in contao) - looks like I was wrong.
So this turns out to be more or less a feature request.

**Comment by fritzmg** [comment]
Ah yes, I missed that. Yeah, `TL_BODY` should get processed in the `BackendTemplate` as well.

**Comment by leofeyer** [comment]
@contao/developers Do you think the back end should support `TL_HEAD` and `TL_BODY`?

**Comment by aschempp** [comment]
Sometimes that would have been handy in the past. But apparently I always found other solutions. 

As it would be a feature, I wonder what would be the best way to add something like that in a Symfony way. I already did some thinking on a "add page output" service but I havent come up with a good way.

**Comment by fritzmg** [comment]
Since `TL_MOOTOOLS` is already processed in the backend template, I'd consider it a bug that `TL_BODY` isn't, not a feature.

**Comment by discordier** [comment]
It is a feature and more over, it alters the current behaviour.
It will now add stuff in the backend that was so far guaranteed to only get added in the frontend.
I wonder if we can call this incompatible change, as this is arguable but it definitely is a feature and not a bug.

**Comment by Fuzzyma** [comment]
This "stuff" would be only added when it's in the array. And it's only in the array if the user put in there. Since it wasn't supported till now I would be surprised if anyone would have added anything to the array when the backend mode is active. 
The only thing I can imagine is the preview of Frontend elements where a custom module is imported or something. But even then it must likely would only add behavior which is related to this module and not break anything

**Comment by fritzmg** [comment]
> And it's only in the array if the user put in there.

Yes, but the user might have put it there for the Frontend only.

**Comment by Fuzzyma** [comment]
When he put it there for frontend only then there would be no problem anyway

**Comment by fritzmg** [comment]
> When he put it there for frontend only then there would be no problem anyway

It could have unintended side effects in the backend.

**Comment by Fuzzyma** [comment]
Thats what I tried to point out.
In most cases the added code just would not work - which is no "bad" side effect itself.
In some cases it will work as it should.
Only in some cases it will interfer with the backend code (e.g. jquery vs mootools) but that is a negligible amount imo.

**Comment by leofeyer** [comment]
> It could have unintended side effects in the backend.

I'm also against changing the current behavior.

**Comment by leofeyer** [comment]
> It is a feature and more over, it alters the current behaviour.

As discussed in Mumble on April 21st, this definitely implies a BC break.
